sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -46,6 +47,16 @@ object OrcUtils extends Logging {
     "ZLIB" -> ".zlib",
     "LZO" -> ".lzo")
 
+  def rawSize(hadoopConf: Configuration, filePath: Path): BigInt = {
+    val fs = filePath.getFileSystem(hadoopConf)
+    val readerOptions = OrcFile.readerOptions(hadoopConf).filesystem(fs)
+    try {
+      Utils.tryWithResource(OrcFile.createReader(filePath, readerOptions))(_.getRawDataSize)
```

Comment:
I'm not faimilar with the ORC impl. though, does this method actually read data to compute its size? cc: @dongjoon-hyun

Reply:
It uses the column statistics of ORC, see [the relevant part of ReaderImpl](https://github.com/apache/orc/blob/bf6fc4ffab026e74557c50de9f2a827ddf535bdb/java/core/src/java/org/apache/orc/impl/ReaderImpl.java#L767-L864)

Reply:
Ah, I see.